packages/fdc3-agent-proxy/src/channels/DefaultChannelSupport.ts

```diff
@@ -1,35 +1,44 @@
+import {
+  Channel,
+  ContextHandler,
+  Listener,
+  PrivateChannel,
+  ChannelSelector,
+  EventHandler,
+  ChannelError,
+} from '@kite9/fdc3-standard';
 import { Messaging } from '../Messaging';
 import { ChannelSupport } from './ChannelSupport';
 import { DefaultPrivateChannel } from './DefaultPrivateChannel';
 import { DefaultChannel } from './DefaultChannel';
 import { DefaultContextListener } from '../listeners/DefaultContextListener';
-import { BrowserTypes } from '@kite9/fdc3-schema';
-import { FollowingContextListener } from '../listeners/FollowingContextListener';
+import { UserChannelContextListener } from '../listeners/UserChannelContextListener';
 import { EventListener } from '../listeners/EventListener';
-
-type ChannelDetail = BrowserTypes.Channel;
-type GetUserChannelsRequest = BrowserTypes.GetUserChannelsRequest;
-type GetUserChannelsResponse = BrowserTypes.GetUserChannelsResponse;
-type GetOrCreateChannelResponse = BrowserTypes.GetOrCreateChannelResponse;
-type GetOrCreateChannelRequest = BrowserTypes.GetOrCreateChannelRequest;
-type CreatePrivateChannelRequest = BrowserTypes.CreatePrivateChannelRequest;
-type CreatePrivateChannelResponse = BrowserTypes.CreatePrivateChannelResponse;
-type JoinUserChannelResponse = BrowserTypes.JoinUserChannelResponse;
-type JoinUserChannelRequest = BrowserTypes.JoinUserChannelRequest;
-type GetCurrentChannelResponse = BrowserTypes.GetCurrentChannelResponse;
-type GetCurrentChannelRequest = BrowserTypes.GetCurrentChannelRequest;
-type LeaveCurrentChannelRequest = BrowserTypes.LeaveCurrentChannelRequest;
-type LeaveCurrentChannelResponse = BrowserTypes.LeaveCurrentChannelResponse;
+import {
+  GetCurrentChannelResponse,
+  GetCurrentChannelRequest,
+  GetUserChannelsResponse,
+  GetUserChannelsRequest,
+  GetOrCreateChannelResponse,
+  GetOrCreateChannelRequest,
+  CreatePrivateChannelResponse,
+  CreatePrivateChannelRequest,
+  LeaveCurrentChannelResponse,
+  LeaveCurrentChannelRequest,
+  JoinUserChannelResponse,
+  JoinUserChannelRequest,
+} from '@kite9/fdc3-schema/generated/api/BrowserTypes';
+import { throwIfUndefined } from '../util';
 
 export class DefaultChannelSupport implements ChannelSupport {
-  protected userChannels: Channel[] | null = null;
-  private followingListeners: FollowingContextListener[] = [];
-
-  constructor(
-    private readonly messaging: Messaging,
-    private readonly channelSelector: ChannelSelector
-  ) {
+  readonly messaging: Messaging;
+  readonly channelSelector: ChannelSelector;
+  protected userChannels: Channel[] = [];
+  protected userChannelListeners: UserChannelContextListener[] = [];
+
+  constructor(messaging: Messaging, channelSelector: ChannelSelector) {
+    this.messaging = messaging;
+    this.channelSelector = channelSelector;
     this.channelSelector.setChannelChangeCallback((channelId: string | null) => {
       if (channelId == null) {
```

Comment:
still not tested

Reply:
I believe this to be the main point outstanding (timeouts and debug logging to go in other PRs) - will ping when I have it solved.

Reply:
Well this and having a CI check fail if one of the builds fails - are you able to look at that @robmoffat?

Reply:
This is untested in the base branch. I think thats because this code has to be called by the fdc3-get-agent package and is exercised in the fdc3-get-agent tests. To test it here we'll need an additional mock for the channel selector - working that up.

Reply:
Added the test for this